packages/core/lib/strategies/DefaultStrategy.ts

```diff
@@ -87,7 +92,7 @@ export class DefaultStrategy implements CacheStrategy {
 
       this.pendingMethodCallMap.set(context.key, methodPromise);
 
-      result = await methodPromise;
```

Comment:
What do you think about changing this bottom part to something like this so the `pendingMethodCallMap` value is cleared regardless of success/failure?

```
      try {
        this.pendingMethodCallMap.set(context.key, methodPromise);
        result = await methodPromise;
      } catch(err) {
        throw err;
      } finally {
        this.pendingMethodCallMap.delete(context.key);
      }
```

Reply:
Good idea! I'll include it.